Vite can not resolve path dependencies

### Describe the bug

We use Vue 3 with Typescript support, script setup and Swiper 7.3.2 in a project.
In our App, we include Swiper Types. If we want to run vite-ts (2.7.1) the following error occurs:

`Pre-bundling dependencies:
vue
bootstrap
vue3-form-validation
@fortawesome/fontawesome-svg-core
vue-router
(...and 15 more)
(this will be run only when your dependencies or config have changed)

    node_modules/swiper/types/index.d.ts:1:14: error: Could not resolve "./shared"
    1 │ export * from './shared';
    ╵ ~~~~~~~~~~

    node_modules/swiper/types/index.d.ts:2:34: error: Could not resolve "./swiper-class"
    2 │ export { default as Swiper } from './swiper-class';
    ╵ ~~~~~~~~~~~~~~~~

    node_modules/swiper/types/index.d.ts:3:14: error: Could not resolve "./swiper-events"
    3 │ export * from './swiper-events';
    ╵ ~~~~~~~~~~~~~~~~~

    node_modules/swiper/types/index.d.ts:4:14: error: Could not resolve "./swiper-options"
    4 │ export * from './swiper-options';
    ╵ ~~~~~~~~~~~~~~~~~~

    node_modules/swiper/types/index.d.ts:5:14: error: Could not resolve "./modules/public-api"
    5 │ export * from './modules/public-api';
    ╵ ~~~~~~~~~~~~~~~~~~~~~~

error when starting dev server:
Error: Build failed with 5 errors:
node_modules/swiper/types/index.d.ts:1:14: error: Could not resolve "./shared"
node_modules/swiper/types/index.d.ts:2:34: error: Could not resolve "./swiper-class"
node_modules/swiper/types/index.d.ts:3:14: error: Could not resolve "./swiper-events"
node_modules/swiper/types/index.d.ts:4:14: error: Could not resolve "./swiper-options"
node_modules/swiper/types/index.d.ts:5:14: error: Could not resolve "./modules/public-api"
at failureErrorWithLog (C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:1493:15)
at C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:1151:28
at runOnEndCallbacks (C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:941:63)
at buildResponseToResult (C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:1149:7)
at C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:1258:14
at C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:629:9
at handleIncomingPacket (C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:726:9)
at Socket.readFromStdout (C:\projects\tschingo\vue\node_modules\vite\node_modules\esbuild\lib\main.js:596:7)
at Socket.emit (events.js:315:20)
at addChunk (internal/streams/readable.js:309:12)`

Expected Behavior

No problems resolving path.

### Reproduction

https://codesandbox.io/s/github/forchel/swiper-test or this repository https://github.com/forchel/swiper-test.git

Project setup: yarn install
Compile: yarn dev or yarn vite

### System Info

```shell
System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
    Memory: 47.13 GB / 63.79 GB
  Binaries:
    Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.1.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (94.0.992.31)
    Internet Explorer: 11.0.19041.1202
```


### Used Package Manager

yarn

### Logs

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/vue-next instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/docs/guide/features.md
+++ b/docs/guide/features.md
@@ -35,6 +35,7 @@ Vite only performs transpilation on `.ts` files and does **NOT** perform type ch
 Vite uses [esbuild](https://github.com/evanw/esbuild) to transpile TypeScript into JavaScript which is about 20~30x faster than vanilla `tsc`, and HMR updates can reflect in the browser in under 50ms.
 
 Use the [Type-Only Imports and Export](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) syntax to avoid potential problems like type-only imports being incorrectly bundled. for example:
+
 ```ts
 import type { T } from 'only/types'
 export type { T }

--- a/docs/guide/features.md
+++ b/docs/guide/features.md
@@ -34,6 +34,12 @@ Vite only performs transpilation on `.ts` files and does **NOT** perform type ch
 
 Vite uses [esbuild](https://github.com/evanw/esbuild) to transpile TypeScript into JavaScript which is about 20~30x faster than vanilla `tsc`, and HMR updates can reflect in the browser in under 50ms.
 
+Use the [Type-Only Imports and Export](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) syntax to avoid potential problems like type-only imports being incorrectly bundled. for example:
+```ts
+import type { T } from 'only/types'
+export type { T }
+```
+
 ### TypeScript Compiler Options
 
 Some configuration fields under `compilerOptions` in `tsconfig.json` require special attention.

